### PR TITLE
block cmd: Use coreapi

### PR DESCRIPTION
**Pull request by magik6k**

### CODE REVIEW

**Comment by magik6k** [comment]
Blocked by a bug in go-path

File: core/commands/block.go [hunk 1]
```diff
@@ -282,38 +235,42 @@ It takes a list of base58 encoded multihashes to remove.
...
 
-			cids = append(cids, c)
-		}
-		ch, err := util.RmBlocks(n.Blockstore, n.Pinning, cids, util.RmBlocksOpts{
-			Quiet: quiet,
-			Force: force,
-		})
+			rp, err := api.ResolvePath(req.Context, p)
+			if err != nil {
+				res.SetError(err, cmdkit.ErrNormal)
```
Comment (Kubuxu) [inline core/commands/block.go]: Should we emit error here too?
What do you think?

Comment (Kubuxu) [reply core/commands/block.go]: Reading original code we probably shouldn't unless `ResolvePath` does a lot more than just `Decode`.

File: core/coreapi/block.go [hunk 2]
```diff
@@ -38,13 +38,29 @@ func (api *BlockAPI) Put(ctx context.Context, src io.Reader, opts ...caopts.Bloc
...
 	pref.Version = 1
 
+	if settings.Codec == "" {
+		if settings.MhType != mh.SHA2_256 || (settings.MhLength != -1 && settings.MhLength != 32) {
+			settings.Codec = "protobuf"
+		} else {
+			settings.Codec = "v0"
+		}
+	}
+
```
Comment (Kubuxu) [inline core/coreapi/block.go]: Should this be in `BlockPutOptions` instead, it seems like it is options handling.

Comment (Kubuxu) [reply core/coreapi/block.go]: This way we separate Options validation from implementation.

File: core/commands/block.go (hunk 1 above)
Comment (magik6k) [reply core/commands/block.go]: It does handle ipns and path traversal too

**Pull request merged**